### ISSUE TITLE
External links

### ISSUE THREAD

**Issue by draialexis** [skiasaurus/tau_manifesto#9]
@skiasaurus 

**I spotted a missing link**. Questions ensued.

I noticed that you found links in `FR` to replace the ones in `ENG`, which is awesome by the way. On line 653 (aka `footnote 10`), the link to the definite integral wolfram page was not included in the `FR` version.

**I assume that's on purpose**, and **I further assume that's because the target webpage is in `ENG`**. Is this correct?

**I'd also like to know what your stance is** in those cases: should we include a webpage in `ENG`, in case the reader wants to learn more by using a translator plug-in; for instance? Would you like me to find a relevant webpage in `FR`? Should we stick to the references Michael approved and keep it as it is?

**Comment by skiasaurus** [comment]
This is tricky. In most cases I did try to find French equivalents for links, but it wasn't always possible—examples include personal web pages like "Tau Before It Was Cool" and the "Futurama Math Page," the Guinness World Records site, and of course the Palais article (which would likely require its own, separate translation project if we wanted a French version :stuck_out_tongue:). I had to do a little bit of digging to remember what my decision-making process was when handling Wolfram MathWorld links. The results are as follows:

There are six links to MathWorld in the original. I changed the first three of these (the Fourier transform, the gamma function, and l'Hôpital's rule) to each one's counterpart on French Wikipedia; the fourth—on hyperspheres—I left intact because Michael makes explicit reference to the source in the running text. The remaining two both appear in footnotes, and are _both_ omitted in my translation—the one in footnote 10 that you point out (definite integrals), _and_ another in footnote 19 (hyperspheres again, specifically their multiple, incompatible definitions). For the life of me, I can't remember why I removed these links, or even whether I did so intentionally.

All that said, I was never in love with the idea of using Wikipedia as a substitute for MathWorld—Michael sprinkled links to Wikipedia throughout the original, but for these six specific cases, he opted for a different source; this tells me that he wanted a non-crowdsourced reference for them, which makes using Wikipedia seem counterproductive. So perhaps it would be better to do as you suggest and leave all six as they are and let the reader use a translator if they want.

Another possibility would be to link directly to an auto-translated version of the page, in a similar way to how I handled the xkcd links. I was originally hesitant to do this because of translation algorithms' inconsistent accuracy, and in fact I only used it for xkcd because those links ultimately aren't very important. Now, though, I'm starting to think maybe it wouldn't be so bad to do the same for math references as well, especially if non-English-speaking readers who click through to them are just going to use an auto-translator anyway.

What are your thoughts on all this?

**Comment by draialexis** [comment]
TLDR: IMO: 
equivalent `FR` page **>** auto-translated `ENG` to `FR` page **>** original `ENG` page **>** alternative `FR` page **>** nothing

> the Palais article (which would likely require its own, separate translation project if we wanted a French version 😛)

You're on, dude! We're gonna need to talk about how to translate "wrong" in that context though, cuz "mauvais" still rubs me the wrong way. Actually, we need to talk about it anyway, given how many times it's cited in the Manifesto. Moving on.

Ultimately, I think some links are there for those who want to learn more, some for fun, and some for citing sources consolidate a point. A part of me wants to treat those differently, depending on what function they serve, but I also think consistency is valuable. I mean, it's tricky. I'm tempted to say that what we do with links for fun doesn't really matter.

When Michael points to a wiki article in English and there's an equivalent wiki article in French, that's a no-brainer. In general, I think it's a good idea to preserve the author's source choices. (So I think a link to a MathWorld page in English is better than one to a Wikipedia page in French, if it reflects the source.) It also seems preferable to include a different source in French rather than omitting the source.

How easy is it to link to an auto-translated version of a page? Isn't it going to depend on browers and plug-ins? If it's doable without an unreasonable amount of effort, and it actually works, I'd say it's probably a good idea.

**Comment by skiasaurus** [comment]
> TLDR: IMO:
> equivalent `FR` page **>** auto-translated `ENG` to `FR` page **>** original `ENG` page **>** alternative `FR` page **>** nothing

+1.

> we need to talk about [how to translate "wrong"] anyway, given how many times it's cited in the Manifesto.

Feel free to create an issue for this.

> Ultimately, I think some links are there for those who want to learn more, some for fun, and some for citing sources consolidate a point. A part of me wants to treat those differently, depending on what function they serve, but I also think consistency is valuable. I mean, it's tricky. I'm tempted to say that what we do with links for fun doesn't really matter.

I think the best policy is always one that has some degree of internal consistency, and that your TL;DR above represents that very well. It also suggests something else that I was already thinking about when I made my last comment, which was that we might be able to use auto-translated versions of some of the other English-language sources whose links I left intact, like personal web pages, Guinness, etc.

> How easy is it to link to an auto-translated version of a page? Isn't it going to depend on browers and plug-ins? If it's doable without an unreasonable amount of effort, and it actually works, I'd say it's probably a good idea.

Not difficult at all. Here's what I did for the xkcd links (after finding a page that had the hand-drawn comic transcribed into computer-readable text):

1. Go to Google Translate and set it up to translate from English to French.
2. Paste the English-language page's URL into the box and click the link that shows up on the right side to view the auto-translated page.
3. Copy and paste the auto-translation's URL into a URL shortener like https://bitly.com. (Without this step, the URL will contain `%` signs, and LaTeX will treat everything after first `%` through the end of the line as a comment.)
4. Use the shortened URL as the first argument of the `\href` command.

**Comment by draialexis** [comment]
Issue created.

Glad we agree, and thanks for the how-to. I'll take care of it soon then.

**Comment by skiasaurus** [comment]
I can also do it when I get home from work if you haven’t gotten to it by then.

**Comment by draialexis** [comment]
If you do that's great. I've got another, urgent proofing job on my lap and my brain is absolutely gonna be fried after I'm done (the topic is agriculture and the EU, and its 55 pages 😭 )^^ so no way I'm getting to those links before I go to sleep.

**Comment by skiasaurus** [comment]
Turns out the link doesn't need to be shortened when using Softcover, so the process is even simpler than I thought:
1. Go to Google Translate and set it up to translate from English to French.
2. Paste the English-language page's URL into the box.
3. Use the target URL of the link that appears on the right as the first argument of your `\href` command.

I'm working on the first couple.

**Comment by skiasaurus** [comment]
> 1. Go to Google Translate and set it up to translate from English to French.
> 2. Paste the English-language page's URL into the box.
> 3. Use the target URL of the link that appears on the right as the first argument of your `\href` command.

... Actually it's even simpler than that: just prepend `https://translate.google.com/translate?sl=en&tl=fr&u=` to the beginning of the URL of the page to be auto-translated. See 10e3fc5 for an example.

**Comment by skiasaurus** [comment]
I've reassigned this to myself so that you can focus on the many other things that I can't do.

There are a couple of links that are less straightforward than the rest; I will open separate issues for those.

**Comment by draialexis** [comment]
Appreciated 🙋‍♂️

**Comment by skiasaurus** [comment]
@draialexis, please make sure to check bbf067f, specifically my attempt to translate "Futurama Math Page."

**Comment by draialexis** [comment]
~"page pour mathématiques à Futurama"~ 
"page de maths de Futurama" ("page de mathématiques de Futurama" if you want to be more formal, but that's probably not necessary)

**Comment by skiasaurus** [comment]
The link here

https://github.com/skiasaurus/tau_manifesto/blob/d4e6e22519518e98b567f25d4d6442bc57eb93b9/chapters/tau_manifesto.tex#L810

is to Google's archive of Usenet posts... and yet, _their own translation product_ apparently can't handle it:

![Screenshot from 2020-09-01 18-52-07](https://user-images.githubusercontent.com/5809600/91913725-4a068180-ec84-11ea-9dee-e0fabc87fb89.png)

The [only other archive](https://sci.math.narkive.com/tkgN4Ypf/pi-2-pi-and-frivolity) of that thread that I can find only includes the last five messages, of which the message Michael links to is not one. So I think we may have to just include the original link and let readers copy-paste the text into a translator.

**Comment by draialexis** [comment]
Yep, that's probably fine.

**Comment by skiasaurus** [comment]
The only part of the essay left for me to go through for this issue is the "About the Author" section.

**Comment by skiasaurus** [comment]
I forgot to put it in the commit message, but c73532b is part of fixing this issue.

**Comment by skiasaurus** [comment]
Fixed as of e968c5e.
